content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md

```diff
@@ -1021,42 +1021,54 @@ Example of a response from a webhook indicating a conversion request failed, wit
 
 ## Writing, reading, and updating versioned CustomResourceDefinition objects
 
-When an object is written, it is persisted at the version designated as the
+When an object is written, it is stored at the version designated as the
 storage version at the time of the write. If the storage version changes,
 existing objects are never converted automatically. However, newly-created
 or updated objects are written at the new storage version. It is possible for an
 object to have been written at a version that is no longer served.
 
-When you read an object, you specify the version as part of the path. If you
-specify a version that is different from the object's persisted version,
-Kubernetes returns the object to you at the version you requested, but the
-persisted object is neither changed on disk, nor converted in any way
-(other than changing the `apiVersion` string) while serving the request.
+When you read an object, you specify the version as part of the path.
 You can request an object at any version that is currently served.
+If you specify a version that is different from the object's stored version,
+Kubernetes returns the object to you at the version you requested, but the
+stored object is not changed on disk.
+
+What happens to the object that is being returned while serving the read
+request depends on what is specified in the CRD's `spec.conversion`:
+- if the default `strategy` value `None` is specified, the only modifications
+  to the object are changing the `apiVersion` string and perhaps [pruning
+  unknown fields](/docs/concepts/extend-kubernetes/api-extension/custom-resources/custom-resource-definitions/#field-pruning)
+  (depending on the configuration). Note that this is unlikely to lead to good
+  results if the schemas differ between the storage and requested version.
+  In particular, you should not use this strategy if the same data is
+  represented in different fields between versions.
+- if [webhook conversion](#webhook-conversion) is specified, then this
+  mechanism controls the conversion.
 
 If you update an existing object, it is rewritten at the version that is
 currently the storage version. This is the only way that objects can change from
 one version to another.
```

Comment:
:thought_balloon: we should explain what happens if you do a no-op server-side apply change using the stored version of that API (eg _stored_ version is v1, SSA request uses v2 (but a no-op), and the storage version for that API is v2).

Readers might want to know:
- does the no-op quality of the request mean that the stored version is unchanged?
- does the storage version change because conversion is required, and a request that causes a write has been admitted
- does something else affect the outcome here?

Reply:
The write short-circuit happens at the last minute by comparing the bytes about to be stored against the bytes already stored.

That means a "no-op" update of an object currently stored in v1 but whose CRD has since changed the storage version to v2 will persist an update in v2 format, even if nothing else changed (because the `apiVersion: .../v2` bytes would be different).